nodejs/eks/cluster.ts

```diff
@@ -579,9 +593,71 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
         nodeSecurityGroupTags: args.nodeSecurityGroupTags,
         storageClasses: userStorageClasses,
         fargateProfile: fargateProfile,
+        oidcProvider: oidcProvider,
     };
 }
 
+// Thumbprint retrieval below adapted from https://git.io/JvGHB.
+
+// Find the root CA cert in a chain of certs.
+function findCACertificate(certificate: tls.DetailedPeerCertificate): tls.DetailedPeerCertificate {
+    let cert = certificate;
+    for (; cert?.issuerCertificate?.fingerprint !== cert?.fingerprint;) {
+        cert = cert.issuerCertificate;
+    }
+    return cert;
+}
+
+const THUMBPRINT_MAX_RETRIES: number = 12;
+const THUMBPRINT_SLEEP_INTERVAL: number = 5000; // In milliseconds.
+
+// Get the URL's issuing CA cert thumbprint.
+// This is used for OIDC provider configuration.
+function getIssuerCAThumbprint(issuerUrl: pulumi.Output<string>): pulumi.Output<string> {
+    return issuerUrl.apply(url => getThumbprint(url, THUMBPRINT_MAX_RETRIES, THUMBPRINT_SLEEP_INTERVAL));
+}
+
+async function getThumbprint(issuerUrl: string, retriesLeft: number, interval: number): Promise<string> {
```

Comment:
Not required - but given how specialized all this code is - may make sense to pull out into another file.  It could also then just export `getIssuerCAThumbprint` which I believe is the only public interface to this functionality.

Reply:
Done